### PR TITLE
Ryan/v2

### PR DESCRIPTION
**Pull request by ryancragun**
My apologies in advance for the massive PR.  This is essentially a complete rewrite that uses https://github.com/chef-partners/packman to dynamically build packer definitions and provisioner files to build the marketplace images.

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -17,155 +17,101 @@
...
-      end
+    def aws_builders
+      aws_products.map { |p| p['name'] }
     end
 
-    def uninstall_marketplace
-      chef_ingredient 'chef-marketplace' do
-        action :uninstall
-      end
+    def enabled_aws_products
```
Comment (adamleff) [inline libraries/helpers.rb]: I feel like a lot of these helpers could be made more dynamic, looping over known sets of clouds, marketplaces, etc. so when it comes time to add another marketplace, we just add it to the list vs. having to create 4-5 more helpers that are relatively static.

Not something we need to do for this version given we don't know what the growth plan looks like, but I could see this as a monotonous unnecessary task to add these helpers :)

Comment (ryancragun) [inline libraries/helpers.rb]: I thought about that but doing it imperatively would require lots of deep fetching magic and logic so I opted for the simpler declarative approach.  I don't expect us to add that many more clouds and in the case that we do, a few more simple methods isn't that rough.  If you have an imperative method that you think will work better I'm all ears.

Comment (adamleff) [inline libraries/helpers.rb]: I have some ideas, but I'm not going to hold this up for that. It's totally fine as-is and when it becomes an annoyance, I'll be happy to address it. :)

File: recipes/_publish.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+marketplace_products.each do |product|
```
Comment (adamleff) [inline recipes/_publish.rb]: This is very delightful to look at. I'll come in and add the GCE-specific provisioner steps we need after this gets merged to master.

**Comment by adamleff** [comment]
Super thrilled to see this work. Nice job, @ryancragun. Can't wait to start using it and get the google stuff in. :+1:

**Pull request merged**